kit/src/test/java/com/oracle/javafx/scenebuilder/kit/skeleton/SkeletonFileNameProposalTest.java

```diff
@@ -0,0 +1,192 @@
+        
+        File result             = classUnderTest.create(fxmlLocation, fxControllerName);
+                
+        File expected           = new File("src/main/java/com/oracle/javafx/scenebuilder/kit/skeleton",
+                                           "SkeletonTestController.java").getAbsoluteFile();
+        
+        assertEquals(expected, result);
+        
+    }
+	
+	@Test
+    public void that_fxmlLocation_in_resources_dir_is_changed_to_kotlin_specific_directory() throws Exception {
+        
+        classUnderTest = new SkeletonFileNameProposal(LANGUAGE.KOTLIN);
+        
+        String fxControllerName = "com.oracle.javafx.scenebuilder.kit.skeleton.SkeletonTest$SkeletonTestController";
+        
+        URL  fxmlLocation       = new File("src/main/resources/com/oracle/javafx/scenebuilder/kit/skeleton/SkeletonWindow.fxml")
+                                      .toURI().toURL();
+        
+        File result             = classUnderTest.create(fxmlLocation, fxControllerName);
+                
+        File expected           = new File("src/main/kotlin/com/oracle/javafx/scenebuilder/kit/skeleton",
+                                           "SkeletonTestController.kt").getAbsoluteFile();
+        
+        assertEquals(expected, result);
+        
+    }
+	
+	@Test
+	public void that_controllerName_is_preferred_over_fxmlLocation_in_user_directory() throws Exception {
+		
+		classUnderTest = new SkeletonFileNameProposal(LANGUAGE.JAVA);
+		
+		String fxControllerName = "com.oracle.javafx.scenebuilder.kit.skeleton.SkeletonTest$SkeletonTestController";
+		
+		URL    fxmlLocation     = new File("not-existing-location/Empty.fxml").toURI().toURL();
+		
+		File   result           = classUnderTest.create(fxmlLocation, fxControllerName);
+		
+		File   expected         = new File(System.getProperty("user.home"), "SkeletonTestController.java"); 
+		
+		assertEquals(expected, result);
+		
+	}
+	
+	@Test
+	public void that_fxmlLocations_is_used_when_controllerName_not_exists() throws Exception {
+		
+		classUnderTest = new SkeletonFileNameProposal(LANGUAGE.JAVA);
+		
+		URL  fxmlLocation       = new File("Empty.fxml").getAbsoluteFile().toURI().toURL();
+		
+		File result             = classUnderTest.create(fxmlLocation, null);
+		
+		File expected           = new File("EmptyController.java").getAbsoluteFile(); 
+		
+		assertEquals(expected.getAbsolutePath(), result.getAbsolutePath());
+		
+	}
+	
+	@Test
+	public void that_incorrectly_named_fxml_can_be_handled() throws Exception {
+		
+		classUnderTest = new SkeletonFileNameProposal(LANGUAGE.JAVA);
+		
+		URL  fxmlLocation       = new File("EmptyFxml").getAbsoluteFile().toURI().toURL();
+		
+		File result             = classUnderTest.create(fxmlLocation, null);
+		
+		File expected           = new File("EmptyFxmlController.java").getAbsoluteFile(); 
+		
+		assertEquals(expected.getAbsolutePath(), result.getAbsolutePath());
+		
+	}
+	
+}
```

Comment:
Replace TABS with SPACES on this file

Reply:
Yes, I fixed this as well. Well, I've now created a preset for the Eclipse Java code formatter. This should help.
This is possibly something, which might be useful to provide in the CONTRIBUTING.md once. I'll see to make a proposal.

Reply:
How about adding a checkstyle rules and fail the Maven build so that these can be tackled during development?